src/csvinputfileconn.cc

```diff
@@ -83,6 +83,7 @@ namespace dd
 	else _cifc->add_train_csvline(std::to_string(_cifc->_csvdata.size()+1),vals);
 	++l;
       }
+    _cifc->update_columns();
```

Comment:
Is this a bug fix ?

Reply:
yes, columns were not updated in case of csv data passed directly in json